File: sunnad-ios/sunnad-ios/Domain/Models/SessionUser.swift
import Foundation

enum AuthProvider: String, Equatable, Sendable {
    case email
    case google
    case apple
    case unknown
}

struct SessionUser: Equatable, Sendable {
    let id: UUID
    let email: String?
    let username: String?
    let avatarURL: URL?
    let provider: AuthProvider

    init(
        id: UUID,
        email: String?,
        username: String?,
        avatarURL: URL? = nil,
        provider: AuthProvider = .unknown
    ) {
        self.id = id
        self.email = email
        self.username = username
        self.avatarURL = avatarURL
        self.provider = provider
    }

    var displayName: String {
        if let username, !username.isEmpty {
            return username
        }

        if let email, !email.isEmpty {
            return email
        }

        return "User"
    }
}

File: sunnad-android/app/src/main/java/com/arystan/almasuly/sunnadandroid/core/model/SessionUser.kt
package com.arystan.almasuly.sunnadandroid.core.model

import java.util.UUID

enum class AuthProvider {
    EMAIL,
    GOOGLE,
    APPLE,
    UNKNOWN
}

data class SessionUser(
    val id: UUID,
    val email: String?,
    val username: String?,
    val avatarUrl: String? = null,
    val provider: AuthProvider = AuthProvider.UNKNOWN
) {
    val displayName: String
        get() = when {
            !username.isNullOrBlank() -> username
            !email.isNullOrBlank() -> email
            else -> "User"
        }
}
